src/deluge/gui/views/instrument_clip_view.cpp

```diff
@@ -4600,9 +4600,10 @@ ActionResult InstrumentClipView::verticalEncoderAction(int32_t offset, bool inCa
 
 			// If shift button not pressed, transpose whole octave
 			if (!Buttons::isShiftButtonPressed()) {
-				// If in scale mode, an octave takes numModeNotes rows while in chromatic mode it takes 12 rows
```

Comment:
As this transpose code is common to both automationView and instrumentClipView, I would suggest factoring out the code to an InstrumentClipView:: function that can be called from automation view so we don't have code duplication.

But maybe that can be a different PR after this one is merged :)

Reply:
In #2189 while thinking about this one still.